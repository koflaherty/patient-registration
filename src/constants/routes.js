export const HOME_PAGE = '/';
export const SUMMARY_PAGE = '/summary';
export const SUBMITTED_PAGE = '/submitted';