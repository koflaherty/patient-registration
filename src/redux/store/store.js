import { createStore, combineReducers } from 'redux';
import { reducer as reduxFormReducer } from 'redux-form';
import { enteredRegistration } from "../reducers/reducers";

export const ENTERED_REGISTRATION_STATE_KEY = 'registration';

const reducer = combineReducers({
    form: reduxFormReducer,
    [ENTERED_REGISTRATION_STATE_KEY]: enteredRegistration,
});

export const store = (window.devToolsExtension
    ? window.devToolsExtension()(createStore)
    : createStore)(reducer);
