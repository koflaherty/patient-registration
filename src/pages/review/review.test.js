import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from "react-redux";
import {store} from "../../redux/store/store";
import {mount} from "enzyme/build/index";
import { MemoryRouter, Link } from 'react-router-dom';
import Review from './review';
import { HOME_PAGE, SUBMITTED_PAGE } from "../../constants/routes";

const defaultMockRegistrationData = {
    firstName: 'Kevin',
    lastName: `O'Flaherty`,
};

const reviewExample = (
    <MemoryRouter>
        <Provider store={store}>
            <Review/>
        </Provider>
    </MemoryRouter>
);

const mockRegistrationData = (data = defaultMockRegistrationData) => {
    const registrationDataMock = jest.spyOn(Review.prototype, "_getRegistrationData");
    registrationDataMock.mockImplementation(() => data);

    return registrationDataMock;
};

it('renders snapshot', () => {
    const mock = mockRegistrationData();

    const tree = renderer
        .create(reviewExample)
        .toJSON();
    expect(tree).toMatchSnapshot();

    mock.mockRestore();
});

describe('summary', () => {
    it('should render keys as words', () => {
        const mock = mockRegistrationData();

        const wrapper = mount(reviewExample);
        const liItems = wrapper.find('li');
        expect(liItems.at(0).text()).toEqual(`First Name: ${defaultMockRegistrationData['firstName']}`);
        expect(liItems.at(1).text()).toEqual(`Last Name: ${defaultMockRegistrationData['lastName']}`);
        expect(liItems.length).toEqual(2);

        mock.mockRestore();
    });

    it('should ignore undesired keys (agreed)', () => {
        const mock = mockRegistrationData({
            agree: true,
        });

        const wrapper = mount(reviewExample);
        const liItems = wrapper.find('li');
        expect(liItems.length).toEqual(0);

        mock.mockRestore();
    });
});

describe('links', () => {

    it('should have a link to home page', () => {
        const wrapper = mount(reviewExample);
        let links = wrapper.find(Link);

        expect(links.at(0).prop('to')).toEqual(HOME_PAGE);
    });

    it('should have a link to submit page', () => {
        const wrapper = mount(reviewExample);
        let links = wrapper.find(Link);

        expect(links.at(1).prop('to')).toEqual(SUBMITTED_PAGE);
    });
});