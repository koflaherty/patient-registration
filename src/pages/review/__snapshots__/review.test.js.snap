// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<div>
  <h1>
    Does this information look correct?
  </h1>
  <div>
    <ul>
      <li>
        First Name: Kevin
      </li>
      <li>
        Last Name: O'Flaherty
      </li>
    </ul>
  </div>
  <a
    href="/"
    onClick={[Function]}
  >
    <button
      className="ant-btn"
      onClick={[Function]}
      style={
        Object {
          "marginRight": "10px",
        }
      }
      type="button"
    >
      <span>
        Edit Info
      </span>
    </button>
  </a>
  <a
    href="/submitted"
    onClick={[Function]}
  >
    <button
      className="ant-btn ant-btn-primary"
      onClick={[Function]}
      type="button"
    >
      <span>
        Complete Registration
      </span>
    </button>
  </a>
</div>
`;
